library/foundation/src/main/java/com/epmedu/animeal/foundation/input/PhoneInputView.kt

```diff
@@ -0,0 +1,155 @@
+                    shape = RoundedCornerShape(12.dp),
+                ),
+            horizontalArrangement = Center,
+        ) {
+            Box(
+                modifier = Modifier
+                    .height(56.dp)
+                    .padding(start = 16.dp),
+                contentAlignment = Alignment.Center,
+            ) {
+                Text(
+                    modifier = Modifier.align(Alignment.Center),
+                    text = "+995",
+                    style = TextStyle(color = Color.Black, fontSize = 16.sp)
+                )
+            }
+            BasicTextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 16.dp)
+                    .align(CenterVertically),
+                value = value,
+                onValueChange = {
+                    if (it.length <= 9) {
+                        onValueChange(it)
+                    }
+                },
+                textStyle = TextStyle(color = Color.Black, fontSize = 16.sp),
+                visualTransformation = PhoneFormatTransformation,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                enabled = enabled
+            )
+        }
+    }
+}
+
+internal object PhoneFormatTransformation : VisualTransformation {
+    private const val PHONE_NUMBER_FORMAT = "xxx xx-xx-xx"
+
+    override fun filter(text: AnnotatedString): TransformedText {
+        val trimmed =
+            if (text.text.length >= 9) text.text.substring(0..8) else text.text
+        val annotatedString = AnnotatedString.Builder().run {
+            for (i in trimmed.indices) {
+                append(trimmed[i])
+                if (i == 2) {
+                    append(" ")
+                } else if (i == 4 || i == 6) {
+                    append("-")
+                }
+            }
+            PHONE_NUMBER_FORMAT.takeLast(PHONE_NUMBER_FORMAT.length - length)
+            toAnnotatedString()
+        }
+        @Suppress("ReturnCount")
+        return TransformedText(
+            text = annotatedString,
+            offsetMapping = object : OffsetMapping {
+                override fun originalToTransformed(offset: Int): Int {
+                    if (offset <= 2) return offset
+                    if (offset <= 4) return offset + 1
+                    if (offset <= 6) return offset + 2
+                    if (offset <= 9) return offset + 3
+                    return 12
+                }
+
+                override fun transformedToOriginal(offset: Int): Int {
+                    if (offset <= 2) return offset
+                    if (offset <= 4) return offset - 1
+                    if (offset <= 8) return offset - 2
+                    if (offset <= 10) return offset - 3
+                    return 9
+                }
+            }
+        )
+    }
+}
+
+@Preview(uiMode = UI_MODE_NIGHT_YES, showBackground = true)
```

Comment:
Please return first preview with default light theme and remove showBackground like this: 
```
@Preview
@Preview(uiMode = UI_MODE_NIGHT_YES)
```